src/main/kotlin/spireMapOverhaul/zones/humility/actions/ApplyPowerAllMonstersAction.kt

```diff
@@ -0,0 +1,31 @@
+package spireMapOverhaul.zones.humility.actions
+
+import basemod.interfaces.CloneablePowerInterface
+import com.megacrit.cardcrawl.actions.AbstractGameAction
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
+import com.megacrit.cardcrawl.core.AbstractCreature
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon
+import com.megacrit.cardcrawl.powers.AbstractPower
+
+class ApplyPowerAllMonstersAction(
```

Comment:
this is in stslib now, right?

Reply:
This one works differently, it takes a power instance rather than a lambda that makes powers. It's used to instrument Spiker to change from gaining thorns itself to applying thorns to all enemies by replacing the ApplyPowerAction with this action.
https://github.com/erasels/SpireMapOverhaul/blob/993d474cde6a5d2aa4f46e8c98ed9f1b93c430ac/src/main/kotlin/spireMapOverhaul/zones/humility/patches/beyond/SpikerBuffAll.kt#L21-L27

Reply:
Haha, what a madlad, alright.